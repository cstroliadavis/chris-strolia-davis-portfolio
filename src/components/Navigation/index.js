import { firstCap } from "../../utils/string-helpers";
import { useState } from "react";

export default function Navigation({ currentSection, setCurrentSection, sections }) {
  const [ menuOpen, setMenuOpen ] = useState(false);
  const vm = {
    menuToggle: () => setMenuOpen(!menuOpen),
    menuVisible: menuOpen ? 'menu-visible' : '',
    menuItems: sections.map((section) => ({
      ...section,
      ref: 'about'===section.name ? '#':`#${ firstCap(section.name) }`,
      key: section.name,
      onClick: () => {
        setCurrentSection(section)
      },
      selected: currentSection.name===section.name ? 'selected':null,
    })),
  }

  return (
    <nav className={ `nav-bar ${ vm.menuVisible }` }>
      <div className="portfolio-owner">Chris Strolia-Davis</div>
      <div className="view-nav-toggle" onClick={ vm.menuToggle }>
        <span className="line"></span>
        <span className="line"></span>
        <span className="line"></span>
      </div>
      <div className="menu-items">
        <ul>
          {
            vm.menuItems.map((item) => (
              <li className={ item.selected } key={ item.key }>
                <a href={ item.ref } onClick={ item.onClick }>{ item.title }</a>
              </li>
            ))
          }
        </ul>
      </div>
    </nav>
  );
}
