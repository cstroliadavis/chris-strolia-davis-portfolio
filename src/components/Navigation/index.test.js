import { render, cleanup, fireEvent } from '@testing-library/react';
import Navigation from './index'

afterEach(cleanup);

const sections = Object.freeze([
  Object.freeze({ name: 'about', title: 'About me' }),
  Object.freeze({ name: 'profile', title: 'Profile' }),
]);

const mockSetCurrentSection = jest.fn();
const props = Object.freeze({
  currentSection: sections[0],
  setCurrentSection: mockSetCurrentSection,
  sections,
});

describe('Navigation', () => {
  it('renders', () => {
    render(<Navigation {...props}/>)
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Navigation {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Sets the corresponding section when menu item is clicked', () => {
    const { getByText } = render(<Navigation {...props} />);

    fireEvent.click(getByText('Profile'));

    expect(mockSetCurrentSection).toHaveBeenCalledTimes(1);
    expect(mockSetCurrentSection).toHaveBeenCalledWith(sections[1]);
  });
});
