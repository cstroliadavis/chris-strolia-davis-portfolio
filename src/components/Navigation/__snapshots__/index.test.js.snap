// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav-bar "
  >
    <div
      class="portfolio-owner"
    >
      Chris Strolia-Davis
    </div>
    <div
      class="view-nav-toggle"
    >
      <span
        class="line"
      />
      <span
        class="line"
      />
      <span
        class="line"
      />
    </div>
    <div
      class="menu-items"
    >
      <ul>
        <li
          class="selected"
        >
          <a
            href="#"
          >
            About me
          </a>
        </li>
        <li>
          <a
            href="#Profile"
          >
            Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
